xmppserver/src/main/java/org/jivesoftware/openfire/disco/IQDiscoInfoHandler.java

```diff
@@ -692,7 +696,19 @@ public boolean hasInfo(String name, String node, JID senderJID) {
 
             @Override
             public DataForm getExtendedInfo(String name, String node, JID senderJID) {
-                return null;
+                try {
```

Comment:
Do I have to update this method like this?

Reply:
Almost. I'm not sure why you wrap this in a try/catch. If it's needed, you must never, _never_ forget to log the exception message. You will learn this after you spend weeks tracking down a bug that was hidden from view because someone didn't log the message. :)

When you're logging the warning about there being more than one dataform, it is good to also log:
- how many dataforms there are
- how the code handles this situation (for example: "Only one of the forms will be returned.")